Plugins/NativeEngine/Source/VertexBuffer.h

```diff
@@ -42,69 +42,20 @@ namespace Babylon
 
         static void BuildInstanceDataBuffer(bgfx::InstanceDataBuffer& instanceDataBuffer, const std::map<bgfx::Attrib::Enum, InstanceInfo>& instances, uint32_t instanceCount);
 
-    private:
-        void Build(uint32_t numVertices);
-
-        class Handle final
-        {
-        public:
-            Handle(Graphics::DeviceContext& deviceContext, std::vector<uint8_t> bytes, bool dynamic, uint16_t byteStride);
-            ~Handle();
-
-            // Copy semantics
-            Handle(const Handle&) = delete;
-            Handle& operator=(const Handle&) = delete;
-
-            // Move semantics
-            Handle(Handle&&) noexcept;
-            Handle& operator=(Handle&&) noexcept;
-
-            void Update(const gsl::span<uint8_t> bytes, uint32_t startVertex);
-
-            void Set(bgfx::Encoder* encoder, uint8_t stream, uint32_t startVertex, uint32_t numVertices, bgfx::VertexLayoutHandle layoutHandle);
-
-        private:
-            Graphics::DeviceContext& m_deviceContext;
-            const uintptr_t m_deviceId{};
-            bool m_dynamic{};
-
-            union
-            {
-                bgfx::VertexBufferHandle m_handle{bgfx::kInvalidHandle};
-                bgfx::DynamicVertexBufferHandle m_dynamicHandle;
-            };
-        };
-
     private:
         Graphics::DeviceContext& m_deviceContext;
+        const uintptr_t m_deviceId{};
 
         std::vector<uint8_t> m_bytes{};
         const bool m_dynamic{};
-        bool m_buildCalled{};
-
-        struct AttributeInfo
-        {
-            bgfx::AttribType::Enum AttribType{};
-            uint32_t ByteOffset{};
-            uint8_t NumElements{};
-            bool Normalized{};
-        };
-
-        uint16_t m_byteStride{};
-        std::map<bgfx::Attrib::Enum, AttributeInfo> m_attributes{};
+        uint32_t m_byteStride{};
 
-        std::optional<Handle> m_handle{};
-
-        struct StreamInfo
+        union
         {
-            const AttributeInfo& Attribute;
-            std::optional<Handle> PromoteToFloatsHandle;
-            uint32_t StartVertexOffset;
-            bgfx::VertexLayoutHandle LayoutHandle;
+            bgfx::VertexBufferHandle m_handle{bgfx::kInvalidHandle};
+            bgfx::DynamicVertexBufferHandle m_dynamicHandle;
```

Comment:
This isn't new code, but the m_dynamic + m_handle/m_dynamicHandle union could collectively be replaced with a single std::variant, right?

Reply:
Using std::variant is really annoying in this case because bgfx uses overloads for handles. We tried using std::variant initially and it was hard to write the code.